File: ocean-ois-app/src/main/scala/ch/usi/itc/ois/app/OISAPP.scala
package ch.usi.itc.ois.app

import javafx.application.Application
import javafx.stage.Stage

import ch.usi.itc.ois.gui.{OisMainInterface, OisInterfaceBuilder}

/**
 * Created by maatary on 20/08/14.
 */

object OISAPP {


  def main(args: Array[String]): Unit = {

    javafx.application.Application.launch(classOf[OISAPP], args: _*)

  }

}



class OISAPP extends Application {


  override def start(primaryStage: Stage): Unit = {



    val oisInterface = OisInterfaceBuilder.BuildInterface(primaryStage)

     oisInterface match {

       case OisMainInterface(x, y, z) => {}
    }

    //TODO InitViewServices


    primaryStage.show
  }


}

File: ocean-ois/src/main/scala/ch/usi/ict/dev/ois/OISApp.java
package ch.usi.ict.dev.ois;

import java.util.concurrent.ConcurrentHashMap;









import ch.usi.ict.dev.AclOverSoapHttpMP.AgentIdentifier;
import ch.usi.ict.dev.ois.gui.OISGui;

public class OISApp {
	
	
	/*public class JfxSplitPaneTest extends Application {
	    @Override
	    public void start(Stage stage) throws Exception {
	        SplitPane split = new SplitPane();
	        StackPane child1 = new StackPane();
	        child1.setStyle("-fx-background-color: #0000AA;");
	        StackPane child2 = new StackPane();
	        child2.setStyle("-fx-background-color: #00AA00;");
	        StackPane child3 = new StackPane();
	        child3.setStyle("-fx-background-color: #AA0000;");
	        split.getItems().addAll(child1, child2, child3);
	        //split.setDividerPositions(0.1f, 0.6f, 0.9f)
	        stage.setScene(new Scene(split, 400, 300));
	        stage.show();
	    }
	}*/
	
	/***
	 * Launch in the order ois, then gui, obviously the network infrastructure will be up and running before. 
	 * TODO Launch the ois from the interface only using a buttom for the matter.
	 * In any case the current situation is not problematic.
	 * TODO removing all the creation dependency, either with guice or factories for a start.
	 * @param args
	 */
	public static void main(String[] args) {
		
		
		if (args.length == 0) {
			System.out.println("no args given");
			return;
		}
		
		
		ConcurrentHashMap<String, AgentIdentifier>  registeredAgentMap = new ConcurrentHashMap<String, AgentIdentifier>();
		
		
		OISGui gui                                          = new OISGui(null, registeredAgentMap);
		
		
		//OISInfrastructureService ois 					    = new OISInfrastructureService(args[0], registeredAgentMap, gui);
		
		//new Thread(ois).start();
		
		
		gui.run();
		
		
	}
	
	
	
	

}
